src/sonar_widget/SonarPlot.cc

```diff
@@ -63,8 +63,15 @@ void SonarPlot::setData(const base::samples::Sonar& sonar)
         }
 
         // add current beam to accumulated scanning sonar data
-        if (sonarData.size())
+        else if (sonarData.size())
```

Comment:
I don't understand that change. The size/motor step change above resets the accumulated data. Are you not losing one beam if you add a else if here ?